.github/workflows/check_missing_dependency_versions.yml

```diff
@@ -0,0 +1,37 @@
+name: 'Check missing dependency versions for all crates to be published'
+on:
+  schedule:
+    # At 10:00am UTC every Monday.
+    # See:
+    # - https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
+    # - https://crontab.guru/
+    - cron: '0 10 * * 1'
+  # Allow manual triggering: <https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch>.
+  workflow_dispatch:
+
+jobs:
+  dependency-review:
+    runs-on: buildjet-4vcpu-ubuntu-2204
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          while read -r line; do
+            # Skip empty lines and comments.
+            [[ -z "$line" || "$line" =~ ^# ]] && continue
+
+            echo "Checking crate $line..."
+            output=$(cargo publish --allow-dirty --quiet --dry-run -p "$line" 2>&1)
+
+            # Check if the output contains the error message we're looking for.
+            #
+            # `cargo publish` may fail for many other reasons, but it'd be too
+            # hard to reason about them all. This is the only one we can expect
+            # to catch reliably.
+            echo "$output" | grep -q "all dependencies must have a version specified when publishing."
+            if [ $? -eq 0 ]; then
+                echo "Error: Found problematic output for crate $line."
+                exit 1
+            fi
+          done < packages_to_publish.txt
```

Comment:
One alternative is to fetch the packages that are not publishable automatically instead of relying on the file. That way, we also force all the packages that are not expected to be published to have a `publish = false`. The following command would do the job

```sh
grep -riLE 'publish\s*=\s*false' **/Cargo.toml | xargs sed -n 's/name\s*=\s*"\(.*\)".*/\1/p'
```

Reply:
I love the idea. What do you think about turning this into a dedicated job, and checking that all and only the crates with `publish = true` are present in the `packages_to_publish.txt` file, so we know it's up to date for sure?

Reply:
One-liner to get all publishable crates: `cargo metadata --format-version 1 --no-deps | jq '.packages[] | select(.publish == null) | .name'`. It's similar to what you shared, but it uses `cargo metadata` output and `jq` so it also supports workspace inheritance logic.

I'll try putting together a GH Actions job!

Reply:
Addressed with the new pre-release check scripts!